### PR TITLE
Present modal with custom transition

### PR DESCRIPTION
**Pull request by mluisbrown**
Changed to make view controller presentable as modal VC using custom transition.
Only made a custom transition for the presentation, not the dismissal, where it's not necessary (uses the existing animation).

### CODE REVIEW

**Comment by RuiAAPeres** [comment]
I was testing this on a real device, and there is a point when you dismiss the ImageViewer, and the entire UI stops receiving touches.

**Comment by RuiAAPeres** [comment]
Ok, so with the Close Button action to dismiss, it works. When you flip to dismiss, the rest of the UI stops responding.

**Comment by mluisbrown** [comment]
Ok sorry, forgot to adjust the swipe to dismiss code to handle VC being presented. Now fixed.

**Comment by RuiAAPeres** [comment]
It's working now. Do you think it would take a lot of work to use transitions for the dismiss? This way we would close the circle.

**Comment by RuiAAPeres** [comment]
@mluisbrown  Btw, do you still need this code?

```
        guard let rootController = self.applicationWindow?.rootViewController else { return }
        
        self.view.frame = UIScreen.mainScreen().bounds
        self.applicationWindow!.addSubview(self.view)
        rootController.addChildViewController(self)
        self.didMoveToParentViewController(rootController)
```

**Comment by mluisbrown** [comment]
Shouldn't be too much work to use transitions for the dismiss too. Regarding that code block, it depends if you want to be maintain the ability to use `show()` instead of `presentViewController`. If not, then we can remove the `show` method altogether (and a few other `if`s where we check if the VC was presented or not)

**Comment by RuiAAPeres** [comment]
I was thinking about adding an extension to a `UIViewController`:

```
extension UIViewController {

   func presentImageViewer(imageViewer: ImageViewer, completion: Void -> Void = {}) {
      self.presentModalViewControlle(imageViewer, completion: completion)
  }
}
```

The thing is, if you don't remove the `show`, then you can't remove this as well:

```
-                    self.view.removeFromSuperview()		                 
-                    self.removeFromParentViewController()
```

By removing the `show`, it makes the inner logic simpler.

**Comment by RuiAAPeres** [comment]
@mluisbrown what's your feeling about the extension on an `UIViewController`?

**Comment by mluisbrown** [comment]
I think it's a great idea.

File: ImageViewer/Source/ImageViewer/ImageViewer.swift [hunk 1]
```diff
@@ -280,7 +285,7 @@ public final class ImageViewer: UIViewController, UIScrollViewDelegate {
         }
     }
     
-    private func showAnimation() {
+    internal func showAnimation(duration: NSTimeInterval, completion: ((Bool) -> Void)?) {
```
Comment (RuiAAPeres) [inline ImageViewer/Source/ImageViewer/ImageViewer.swift]: you can remove `internal`, it is by default.

File: ImageViewer/Source/ImageViewer/ImageViewer.swift [hunk 2]
```diff
@@ -323,7 +329,7 @@ public final class ImageViewer: UIViewController, UIScrollViewDelegate {
         }
     }
     
-    private func closeAnimations() {
+    internal func closeAnimation(duration: NSTimeInterval, completion: ((Bool) -> Void)?) {
```
Comment (RuiAAPeres) [inline ImageViewer/Source/ImageViewer/ImageViewer.swift]: Same here

File: README.md [hunk 3]
```diff
@@ -51,7 +51,6 @@ imageViewer.show()
 Because `ImageViewer` was created with a given configuration in mind, it might be limiting factor for certain apps:
 
 * Currently the library will only behave correctly in apps that have rotation disabled (only Portrait). Since we are applying transformations and listening for `UIDeviceOrientationDidChangeNotification`. We have a couple of ideas on how to solve this problem and provide a more predictable behaviour. Given all this,  you shouldn't use for an iPad app.
-* `ImageViewer` is currently a `UIViewController` subclass, we are considering making it a `UIView`, as we find the later lifecycle more reliable. We are adding `ImageViewer`'s root view to the `UIWindow`'s `subViews` and itself as a `childViewController` of the `window.rootViewController`. We are still looking into a way of making this part a bit more idiomatic, while maintaining the great fullscreen look. 
 * We are seeing some issues with the animations due to different aspect ratio between the `displacedView` and the fullscreen `UIImageView`. We aren't sure if it's worth to fix this, as we don't run into this problem.
```
Comment (RuiAAPeres) [inline README.md]: For historical reasons, could you strikethrough this block of text, instead of removing.

**Comment by RuiAAPeres** [comment]
A small bug with the status bar:
status

**Comment by mluisbrown** [comment]
Made changes as per your suggestions. Status bar jump is fixed, but in-call status bar behaviour remains awful. I couldn't find a solution. Using `windowLevel` doesn't work either and is actually worse (ImageView window shifts down with no visible status bar)

**Pull request merged**